packages/doenetml-prototype/src/renderers/doenet/graph.tsx

```diff
@@ -98,9 +119,70 @@ export const Graph: BasicComponent = ({ node }) => {
         );
         setYaxis(yaxis);
 
+        board.on("boundingbox", () => {
+            if (!settingBoundingBox.current) {
+                let newBoundingBox = board.getBoundingBox();
+                let [xMin, yMax, xMax, yMin] = newBoundingBox;
+
+                // look for a change in bounding box that isn't due to round-off error
+                let xScale = Math.abs(xMax - xMin);
+                let yScale = Math.abs(yMax - yMin);
+                let diffs = newBoundingBox.map((v, i) =>
+                    Math.abs(v - (previousBoundingBox.current?.[i] ?? 0)),
+                );
+                if (
+                    Math.max(
+                        diffs[0] / xScale,
+                        diffs[1] / yScale,
+                        diffs[2] / xScale,
+                        diffs[3] / yScale,
+                    ) > 1e-12
+                ) {
+                    previousBoundingBox.current = newBoundingBox;
+
+                    let action: Action = {
+                        component: "graph",
+                        actionName: "changeAxisLimits",
+                        componentIdx: node.data.id,
+                        args: {
+                            x_min: xMin,
+                            x_max: xMax,
+                            y_min: yMin,
+                            y_max: yMax,
+                        },
+                    };
+                    dispatch(coreActions.dispatchAction(action));
+                }
+            }
+        });
+
         setBoard(board);
+
+        return () => {
+            board.off("boundingbox");
+        };
     }, [boardRef]);
 
+    let boundingBox: BoundingBox = [
+        node.data.props.xMin,
+        node.data.props.yMax,
+        node.data.props.xMax,
+        node.data.props.yMin,
+    ];
+
+    if (
+        board &&
+        boundingBox.some((v, i) => v !== previousBoundingBox.current?.[i])
+    ) {
+        settingBoundingBox.current = true;
```

Comment:
Why do you need this ref? This code is running synchronously.

Reply:
This code has been through quite a bit of hardening, but unfortunately with no automated testing (as I couldn't figure out how to get cypress to interact with jsxgraph). I'm loath to change it because I remember having some case that was fixed with this check. Is it possible that jsxgraph does something asynchronously? I remember that making the check that `settingBoundingBox.current` is `false` in `board.on("boundingbox")` stopped some kind of bad behavior.

Reply:
OK. Well, I removed it, and I can't find any adverse effects. I'll just imagine that I changed multiple variables at once when adding in this check however many years ago, and any correlation with behavior was coincidental.